gpflow/base.py

```diff
@@ -32,6 +32,15 @@ def parameters(self):
     def trainable_parameters(self):
         return tuple(self._flatten(predicate=_IS_TRAINABLE_PARAMETER))
 
+    def _repr_html_(self):
```

Comment:
What if we would make just `gpflow.print_summary()` alias function instead of `gpflow.utilities.print_summary()`. People do `import gpflow` anyway, the call will be shorter.

Reply:
@jameshensman ^^^

Reply:
That doesn't solve the issue that this PR addresses, @awav, which is that GPflow objects hould play nice in jupyter environments

Reply:
@jameshensman, it does work for notebooks as well, you can set the format output, globally and in the function itself.

Reply:
@jameshensman, @st-- the PR duplicated what implemented here https://github.com/GPflow/GPflow/blob/develop/gpflow/utilities/utilities.py#L132. It also introduces a circular module dependency between utilities and base classes.

Reply:
print_summary() doesn't address that in IPython or a jupyter notebook, _evaluating an object_ (i.e. just `model` at the end of a cell) should play nicely.

Circular dependency: That's why the imports are only _inside_ the `_repr_html_` and `_repr_pretty_` methods.

Code duplication: We could remove `print_summary()` entirely, add IPython as a dependency, and tell people to use `IPython.display()` instead. `print_summary()` might still be valuable if there is a need for plotting Parameters inside something that subclasses `tf.Module` directly, not `gpflow.base.Module`.

(Should we expose `gpflow.base.Module` in the `gpflow` namespace directly - i.e. `gpflow.Module`?)